gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/BaseSystemTest.java

```diff
@@ -0,0 +1,431 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.AsyncPage;
+import com.google.cloud.Page;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * A base class for system tests. This class can be extended to run system tests in different
+ * environments (e.g. local emulator or remote Pub/Sub service).
+ */
+public abstract class BaseSystemTest {
+
+  /**
+   * Returns the Pub/Sub service used to issue requests. This service can be such that it interacts
+   * with the remote Pub/Sub service (for integration tests) or with an emulator
+   * (for local testing).
+   */
+  protected abstract PubSub pubsub();
+
+  /**
+   * Formats a resource name for testing purpose. For instance, for tests against the remote
+   * service, it is recommended to append to the name a random or time-based seed to prevent
+   * name clashes.
+   */
+  protected abstract String formatForTest(String resourceName);
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mziccard/gcloud-java/file/2584910451/issues/source?bid=3123109&fileBranchId=3304002#l62)